### PR TITLE
Added UT for Reset cart controller.

### PR DESCRIPTION
**Pull request by yashgit891**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: Test/Unit/ResetCartControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            \Magento\Framework\Message\ManagerInterface::class
+        );
+        $this->resultFactoryMock = \Mockery::mock(
+            Magento\Framework\Controller\ResultFactory::class
+        )->makePartial()->shouldAllowMockingProtectedMethods();
+        $this->serializerJson = new \Magento\Framework\Serialize\Serializer\Json;
+        $this->translateInline = \Mockery::mock(
+            \Magento\Framework\Translate\InlineInterface::class
+        );
+        $this->json = new Json($this->translateInline,
```
Comment (garimaprachi) [inline Test/Unit/ResetCartControllerTest.php]: Either on the same line or directly below JSON bracket

File: Test/Unit/ResetCartControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        
+        $this->quoteModel->shouldReceive('load')->andReturn($this->quoteModel);
+        $this->quoteModel->shouldReceive('setIsActive')->andReturn($this->quoteModel);
+        $this->quoteModel->shouldReceive('save')->andReturn($this->quoteModel);
+        
+        $this->messageManager->shouldReceive('addError');
+        
+        $this->resultFactoryMock->shouldReceive('create')->with('json')->andReturn($this->json);
+
+        $this->resetCart = \Mockery::mock(Razorpay\Magento\Controller\Payment\ResetCart::class, [ $this->context,
```
Comment (garimaprachi) [inline Test/Unit/ResetCartControllerTest.php]: Remove space after [

File: Test/Unit/ResetCartControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                                                                                                $this->customerSession,
+                                                                                                $this->checkoutSession,
+                                                                                                $this->checkoutFactory,
+                                                                                                $this->config,
+                                                                                                $this->catalogSession,
+                                                                                                $this->logger])->makePartial()->shouldAllowMockingProtectedMethods();
+        $this->resetCart->objectManager = $this->objectManager;
+    }
```
Comment (garimaprachi) [inline Test/Unit/ResetCartControllerTest.php:109]: Add line after every function end

File: Test/Unit/ResetCartControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        $reflection = new \ReflectionClass($object);
+        $property = $reflection->getProperty($propertyName);
+        $property->setAccessible(true);
+        return $property->getValue($object);
+    }
+    public function testExecuteSuccess()
+    {
+        $this->checkoutSession->shouldReceive('getLastQuoteId')->andReturn('000012');
+        $response = $this->resetCart->execute();
+        $expected_response = '{"success":true,"redirect_url":"checkout\/#payment"}';
```
Comment (garimaprachi) [inline Test/Unit/ResetCartControllerTest.php:127]: add space after comma

File: Test/Unit/ResetCartControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        $this->assertSame($expected_response, $this->getProperty($response, 'json'));
+
```
Comment (garimaprachi) [inline Test/Unit/ResetCartControllerTest.php]: Remove blank line

File: Test/Unit/ResetCartControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        $response = $this->resetCart->execute();
+        $expected_response = '{"success":true,"redirect_url":"checkout\/#payment"}';
+        $this->assertSame($expected_response, $this->getProperty($response, 'json'));
+       
+    }
+    public function testExecuteFailure()
+    {
+        $this->checkoutSession->shouldReceive('getLastQuoteId')->andReturn(null);
+        $response = $this->resetCart->execute();
+        $expected_response = '{"success":true,"redirect_url":"checkout\/cart"}';
```
Comment (garimaprachi) [inline Test/Unit/ResetCartControllerTest.php:138]: Add space after comma

File: Test/Unit/ResetCartControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        $this->assertSame($expected_response, $this->getProperty($response, 'json'));
+       
+    }
+}
```
Comment (garimaprachi) [inline Test/Unit/ResetCartControllerTest.php]: add line after eof

**Review by garimaprachi — COMMENTED** [review]
LGTM

**Review by garimaprachi — COMMENTED** [review]
LGTM

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

**Pull request merged**